### ISSUE TITLE
500 internal server error when SvelteKit returns 301

### ISSUE THREAD

**Issue by saym97** [geoffrich/svelte-adapter-azure-swa#15]
I have been experimenting with my web app and couldn't figure out why one particular SSR route was giving 500 error. Then i started debugging the render function code. I found out that when a route has a sub route for example '/setting/profile', there is trailing slash put at the end of the route like '/setting/' or '/setting/profile/' which makes it a redirection route, so sveltekit keeps on redirecting to same route causing 500 error. 

### Reproducing error on demo repo to be sure: 
if go to demo Todo list app and the put the '/' at the end of /todo route , 
https://delightful-bush-01bdb9a0f.azurestaticapps.net/todos/  (this will give 500).

**Comment by geoffrich** [comment]
Thanks for the report and PR! I want to dig into this a bit more before merging the PR -- I don't think removing the trailing slash would be correct in all cases, since it technically refers to a different route. I wonder if there's an underlying bug with Azure SWA here. But this is a very clear issue report, so it should be easy to repro.

**Comment by saym97** [comment]
I think the way it is getting the original Url from the header, **( headers["x-ms-original-url"];)** is the problem because the trailing slash exist in the header even before it's passed to render function. 
If removing trailing slash is not the best solution then another solution could be passing the original route/url as an input binding to the render function context, if that's even possible?

**Comment by geoffrich** [comment]
Okay, I've dug into this a bit more and it does seem to be an Azure SWA issue, not a SvelteKit or adapter issue. The original URL in the header is correct -- when you navigate to `/todos/`, `x-ms-original-url` _should_ include the trailing slash, since that's in the original URL. By default, SvelteKit will redirect `/todos/` to `/todos` and return a 301 response. However, when the Azure function that performs SSR returns a 301 status, a 500 error page is shown instead. I didn't see any redirect loops; it seems that simply returning a 301 shows the error.

This issue isn't isolated to this adapter. I was able to repro it with a [simple Azure function](https://github.com/geoffrich/azure-swa-redirect-issue/blob/master/api/redirect/index.js) that just returns a 301 and does not invoke SvelteKit. I've opened an issue on the SWA repo for them to address this: https://github.com/Azure/static-web-apps/issues/680

While the fix you proposed in your PR (removing the trailing slash) does mitigate the issue, it won't work in all cases. It's possible to [configure SvelteKit](https://kit.svelte.dev/docs#configuration-trailingslash) to do the opposite of the default and redirect `/todos` to `/todos/`. With that configuration, we'd see the same issue navigating to `/todos`, since SvelteKit would return a 301 response and trigger the error. So I don't think that's the best change going forward. If you do want that behavior in your own app, it would be better to implement it in the [handle hook](https://kit.svelte.dev/docs#hooks-handle).

Do these findings match what you're seeing?
